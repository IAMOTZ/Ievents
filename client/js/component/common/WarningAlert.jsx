import React from 'react';

export default (props) => {
  if (props.message) {
    return (
      <div className="bg-danger text-white text-center p-1 rounded">
        <i class="fa fa-times"></i>&nbsp;
        <span>{props.message}</span>
      </div>
    )
  }
  else {
    return null;
  }
};
