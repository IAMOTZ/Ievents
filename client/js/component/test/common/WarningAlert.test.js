import React from 'react';
import renderer from 'react-test-renderer';
import WarningAlert from '../../common/WarningAlert.jsx';

describe('<WarningAlert />', () => {
  it('should render nothing if no message was passed in', () => {
    const wrapper = shallow(<WarningAlert />);
    expect(wrapper.html()).toBeNull();
  });
  it('should render the component if a message was passed in', () => {
    const tree = renderer.create(<WarningAlert message="Testing WarningAlert" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
