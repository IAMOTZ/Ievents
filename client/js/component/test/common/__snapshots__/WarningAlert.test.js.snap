// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WarningAlert /> should render the component if a message was passed in 1`] = `
<div
  className="bg-danger text-white text-center p-1 rounded"
>
  <i
    className="fa fa-times"
  />
  Â 
  <span>
    Testing WarningAlert
  </span>
</div>
`;
